### DISCUSSION TITLE
Customizing ggcuminc graphs with ggsurvfit

### DISCUSSION THREAD

**Discussion by eloccodes** [pharmaverse/ggsurvfit#138] [Q&A] [answered]
Hi there -- thanks for such a lovely R package. Was wondering how to further customize ggcuminc graphs with ggplot2 language. See reprex and the example below. Thank you! 

In this example, I am graphing the incidence of locoregional failure in a cancer population with death as a competing risk factor. In sum, I would like to be able to change the color and line type of each of my levels on the graph, as well as the titles, colors, etc. of the legend and axes. 

## Load libraries
library(tidyverse) # organizing and cleaning data
library(ggsurvfit) # graphing CI curves
library(tidycmprsk) # competing risk cumulative incidence

## Create data frame with sample data
df <- data.frame(
  ttprog = rexp(100),
  status = factor(
    sample(0:2, 100, replace=TRUE), 
    0:2, 
    c('censored', 'progression', 'death without progression')
  ),
  variable = factor(sample(1:2,  100,replace=TRUE),
                   1:2,
                   c('TrtA','TrtB')
  ))

## Create cuminc function 
test <- cuminc(Surv(ttprog, status) ~ 1, data = df) 

• Failure type "death without progression"
time   n.risk   estimate   std.error   95% CI          
1.00   39       0.150      0.039       0.084, 0.235    
2.00   14       0.290      0.056       0.186, 0.403    
3.00   7        0.290      0.056       0.186, 0.403    

• Failure type "progression"
time   n.risk   estimate   std.error   95% CI          
1.00   39       0.300      0.050       0.206, 0.400    
2.00   14       0.433      0.060       0.315, 0.545    
3.00   7        0.479      0.063       0.352, 0.596  

## ggcuminc which produces a black-and-white gray curve 
test %>% 
  ggcuminc(outcome = c("progression","death without progression")) + 
  add_confidence_interval() +
  add_risktable() +
  add_censor_mark() + 
  labs(title = "Incidence of LRF or Death", x = "Time (months)") +
  theme_bw() + 
  theme(legend.position = "right")

![image](https://user-images.githubusercontent.com/55251204/209839724-95c1140a-ae28-4181-9541-cbc9e2019159.png)

## however, color argument in the confidence interval works

test %>% 
    ggcuminc(outcome = c("progression","death without progression"), color = c("#54738E", "#82AC7C")) + 
    add_confidence_interval() +
    add_risktable() +
    add_censor_mark() + 
    labs(title = "Incidence of LRF or Death", x = "Time (months)") +
    theme_bw() + 
    theme(legend.position = "right")
Error in `ggplot2::geom_step()`:
! Problem while setting up geom aesthetics.
ℹ Error occurred in the 1st layer.
Caused by error in `check_aesthetics()`:
! Aesthetics must be either length 1 or the same as the data (202)
✖ Fix the following mappings: `colour`
Run `rlang::last_error()` to see where the error occurred.

test %>% 
    ggcuminc(outcome = c("progression","death without progression")) + 
    add_confidence_interval(fill = "#508050") +
    add_risktable() +
    add_censor_mark() + 
    labs(title = "Incidence of LRF or Death", x = "Time (months)") +
    theme_bw() + 
    theme(legend.position = "right")

![image](https://user-images.githubusercontent.com/55251204/209840834-216e80ae-27a6-4e9f-aef7-87e0d8b2152a.png)

**Comment by ddsjoberg** [comment]
Check out the first example in the figure gallery.
http://www.danieldsjoberg.com/ggsurvfit/articles/gallery.html

It provides an example of changing the colors.

```r
gg_default <-
  [survfit2](http://www.danieldsjoberg.com/ggsurvfit/reference/survfit2.html)([Surv](https://rdrr.io/pkg/survival/man/Surv.html)(time, status) ~ surg, data = df_colon) [%>%](https://magrittr.tidyverse.org/reference/pipe.html)
  [ggsurvfit](http://www.danieldsjoberg.com/ggsurvfit/reference/ggsurvfit.html)() +
  [add_confidence_interval](http://www.danieldsjoberg.com/ggsurvfit/reference/add_confidence_interval.html)() +
  labs(title = "Default")

gg_styled <-
  gg_default +
  coord_cartesian(xlim = [c](https://rdrr.io/r/base/c.html)(0, 8)) +
  scale_y_continuous(
    limits = [c](https://rdrr.io/r/base/c.html)(0, 1),
    labels = scales::[percent](https://scales.r-lib.org/reference/percent_format.html), 
    expand = [c](https://rdrr.io/r/base/c.html)(0.01, 0)
  ) +
  scale_x_continuous(breaks = 0:9, expand = [c](https://rdrr.io/r/base/c.html)(0.02, 0)) +
  scale_color_manual(values = [c](https://rdrr.io/r/base/c.html)('#54738E', '#82AC7C')) +
  scale_fill_manual(values = [c](https://rdrr.io/r/base/c.html)('#54738E', '#82AC7C')) +
  theme_minimal() +
  theme(legend.position = "bottom") +
  guides(color = guide_legend(ncol = 1)) +
  labs(
    title = "Modified",
    y = "Percentage Survival"
  )

gg_default + gg_styled
```

![image](https://user-images.githubusercontent.com/26774684/209841911-c1de8384-bb87-4669-b166-ca91fc7de032.png)

**Reply by eloccodes** [reply]
Thank you so much for the quick reply! I have definitely pursued the gallery and couldn't get the code to work for me specifically with the **ggcuminc** function. For example for the exact code from the gallery example: 

`test %>%
    ggcuminc(outcome = c("progression","death without progression")) +
    add_confidence_interval() +
    add_risktable() +
    add_censor_mark() +
    labs(title = "Incidence of LRF or Death", x = "Time (months)") +
    coord_cartesian(xlim = c(0, 4)) +
    scale_y_continuous(
        limits = c(0, 1),
        labels = scales::percent, 
        expand = c(0.01, 0)
    ) +
    scale_x_continuous(breaks = 0:4, expand = c(0.02, 0)) +
    scale_color_manual(values = c('#54738E', '#82AC7C')) +
    scale_fill_manual(values = c('#54738E', '#82AC7C')) +
    theme_minimal() +
    theme(legend.position = "bottom") +
    guides(color = guide_legend(ncol = 1))`

It still comes out as black and white.  However, the scale modifications and other code work, wonderfully! 
I have tried other ggplot language without avail, as well. Sorry for such a bother--really appreciate the help and great package! 

![image](https://user-images.githubusercontent.com/55251204/209850445-e5029874-8164-48c8-b085-9d28600dc243.png)

**Comment by ddsjoberg** [comment, marked as answer]
The code from the gallery works for me:

``` r
library(ggsurvfit)
#> Loading required package: ggplot2
library(tidycmprsk)

cuminc(Surv(ttdeath, death_cr) ~ trt, trial) %>%
  ggcuminc(outcome = "death from cancer") +
  add_confidence_interval() +
  add_censor_mark() +
  add_risktable() +
  theme_bw() +
  scale_ggsurvfit(x = list(breaks = seq(0, 24, by = 6))) +
  scale_color_manual(values = c('#54738E', '#82AC7C')) +
  scale_fill_manual(values = c('#54738E', '#82AC7C'))
```

![](https://i.imgur.com/AXWmRQ5.png)

<sup>Created on 2022-12-28 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Reply by ddsjoberg** [reply]
If you're still having issues, run your code piece by piece to identify where the error is arising.

If you would like to share more code, please review the reprex package and make a proper reprex to copy and paste into the discussion. Also be sure to keep your code as minimal as possible, aka only include the most minimal code needed to illustrate an issue.

**Comment by saskia-c** [comment]
I have the same question on customizing lines in the ```ggcuminc()``` function. 

The gallery code works to change the line colours if there are two groups (e.g. Drug A and Drug B). However, if there is only one group, but you would like to plot two (or more) outcomes, then it is not clear how to change the line colours.

Below examples using 'trial' dataset 

``` 
library(tidycmprsk)

# Was expecting this to work, but it returns a black and white plot

cuminc(Surv(ttdeath, death_cr) ~ 1, trial) %>%
  ggcuminc(outcome = c("death from cancer", "death other causes")) +
  add_confidence_interval() +
  scale_color_manual(values = c('#54738E', '#82AC7C')) +
  scale_fill_manual(values = c('#54738E', '#82AC7C'))


# Alternatively thought this might work, but it returns same error as described by eloccodes

cuminc(Surv(ttdeath, death_cr) ~ 1, trial) %>%
  ggcuminc(outcome = c("death from cancer", "death other causes"),
                   color = c('#54738E', '#82AC7C')) +
  add_confidence_interval() 
```

Many thanks in advance!

**Reply by ddsjoberg** [reply]
FYI, here is a solution from stackoverflow: https://stackoverflow.com/questions/79388291
![image](https://github.com/user-attachments/assets/d42f6282-929a-43ec-936f-2c0c01d2b63a)

**Comment by ddsjoberg** [comment]
Hmmm, I am not sure it's possible. The aes is constructed with `aes(x = time, y = estimate, linetype = outcome)`. The colour aes is added if the model is stratified.

I am not sure if colors can be modified if there is no call to a color aes?

**Reply by saskia-c** [reply]
Oke, thank you for looking into it!
